service/collector/cluster_test.go

```diff
@@ -0,0 +1,246 @@
+package collector
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	capiv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	clientfake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	applicationv1alpha1 "github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/k8sclient/v6/pkg/k8sclienttest"
```

Comment:
Ah OK I think this is why we need to use `v7` of k8sclient. Good to upgrade it anyway.